core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java

```diff
@@ -387,6 +389,25 @@ protected DiffRepository getDiffRepos() {
         Collections.emptyList(), Collections.emptyList());
   }
 
+  @Test public void testHintsPropagateWithDifferentKindOfRels() {
+    final String sql = "select /*+ AGG_STRATEGY(TWO_PHASE) */\n"
+        + "ename, avg(sal)\n"
+        + "from emp group by ename";
+    final RelNode rel = tester.convertSqlToRel(sql).rel;
+    final RelHint hint = RelHint.of(
+        Collections.emptyList(),
+        "AGG_STRATEGY",
+        Collections.singletonList("TWO_PHASE"));
+    // Validate Hep planner.
+    HepProgram program = new HepProgramBuilder()
+        .addRuleInstance(AggregateReduceFunctionsRule.INSTANCE)
```

Comment:
Maybe write a mock rule (similar to `MockJoin`) to simulate the `Project + Aggregate` split that you want to test?


I had a hard time to understand why `AggregateReduceFunctionsRule` will trigger what expects be tested?

Reply:
The aim is to test the real rules can propagate the hints successfully. If you also find any bad case, welcome to contribute more.

Reply:
gotcha. Thanks for your clarification.